### ISSUE TITLE
React.AspNet is not working with ASP.NET Core 1.0 RC2

### ISSUE THREAD

**Issue by ShikiGami** [reactjs/React.NET#268]
When using React.AspNet in a ASP.NET Core 1.0 RC2 project with a "net451" framework, it gives the error: 
`The dependency React.AspNet 2.3.0 does not support framework .NETFramework,Version=v4.5.1`

**Comment by Daniel15** [comment]
Looks like I need to change "dnx451" to something else, although the document explaining these identifiers is super confusing: https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md

Basically I want the package to be compatible with .NET Framework 4.5+ but NOT with .NET Core (as some of the dependencies don't run on .NET Core). "dnx451" used to do that, but I have no idea which one to use now.

**Comment by ShikiGami** [comment]
It is "net451", but with that alone isn't enough to make it work on RC2, since there are a lot of structural changes. I actually made a fork and was able to make it compile and I was able to make React.Sample.Mvc6 run on IIS Express, but for some reason Razor doesn't seem to find React.Core assemblies from React.AspNet, the out url became a mess, and it doesn't seem to be creating the nuget package for React.AspNet





差出人: Daniel Lo Nigro

送信日時: 5月20日金曜日 午前9:14

件名: Re: [reactjs/React.NET] React.AspNet is not working with ASP.NET Core 1.0 RC2 (#268)

宛先: reactjs/React.NET

CC: Shiki Byakko, Author



Looks like I need to change "dnx451" to something else, although the document explaining these identifiers is super confusing: https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md


Basically I want the package to be compatible with .NET Framework 4.5+ but NOT with .NET Core (as some of the dependencies don't run on .NET Core). "dnx451" used to do that, but I have no idea which one to use now.

**Comment by ShikiGami** [comment]
Just an update, I was able to make React.Sample.Mvc6 run, but it is giving me an unhandled exception first on `var resultJson = engine.CallFunction<string>(function, args);` from `JavaScriptEngineUtils.cs`
After that it seems to run fine.

Since now on RC2 xproj doesn't seem to be creating a nupkg file from a full .NET Framework class library, I transformed into a csproj and tried to make the nupkg the same way the other projects do it, but it seems that there is a bug in the current version of NuGet that gives you an error when you use RC2 Core libraries inside a full .Net Framework project, so still no package.

For more info on this bug take a look at https://github.com/NuGet/Home/issues/2754

**Comment by Daniel15** [comment]
What's the exception?

For the xproj not creating a NuGet package, it's probably worth creating a
brand new .NET Core class library and see if that produces a NuGet package
on build. If so, diff the xproj files and see what differs.

**Comment by ShikiGami** [comment]
The exception I'm getting is 
```
An unhandled exception of type 'System.AccessViolationException' occurred in MsieJavaScriptEngine.dll

Additional information: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
```
Right now there doesn't seem to be any way to get xproj to create a NuGet package by itself, but I found out there is a new command called `dotnet pack` to create packages out of project.json, so I restored React.AspNet as a xproj, and was able to create the packages by adding a postcompile script at project.json, so it is now working, the only problem is the exception which I have no idea what this all about.

**Comment by Daniel15** [comment]
Thanks for investigating! Could you please send a pull request with your changes, so I can take a look and merge them in? :smile:

**Comment by Daniel15** [comment]
Fixed with #271 - Will try to push to NuGet soon.
